src/datastation/dv_dataset_edit_metadata.py

```diff
@@ -0,0 +1,82 @@
+import csv
+import re
+from argparse import ArgumentParser
+from csv import DictReader
+from os.path import expanduser, isfile
+
+from datastation.common.batch_processing import BatchProcessor
+from datastation.common.config import init
+from datastation.common.utils import add_batch_processor_args, add_dry_run_arg
+from datastation.dataverse.datasets import Datasets
+from datastation.dataverse.dataverse_client import DataverseClient
+
+
+def main():
+    config = init()
+    parser = ArgumentParser(description='Edits one or more, potentially published, datasets. Requires an API token.')
+    parser.add_argument('-r', '--replace', dest="replace", action='store_true',
+                        help="Replace existing metadata fields with the new metadata. ")
+    parser.add_argument('pid_or_file',
+                        help="Either a CSV file or the PID of the dataset to edit. "
+                             "One column column of the file MUST have title 'PID'. "
+                             "The other columns MUST have a typeName, as for the --value argument.")
+    parser.add_argument('-v', '--value', action='append',
+                        help="At least once in combination with a PID, none in combination with a CSV file. "
+                             "The new values for fields formatted as <typeName>=<value>. "
+                             "for example: title='New title'. "
+                             "A subfield in a compound field must be prefixed with "
+                             "the typeName of the compound field and an @ sign, for example: "
+                             "--value author@authorName='the name' "
+                             "--value author@authorAffiliation='the organization'. "
+                             "The quoting style for repetitive fields is <typeName>='[\"<value>\"]', "
+                             "for example: -v dansRightsHolder='[\"me\",\"O'\"'\"'Neill\"]', or in a CSV: "
+                             "'[\"me\",\"O''Neill\"]'. "
+                             "Note that without 'replace' an existing value of a repetitive field is not duplicated. "
+                             "Not repetitive fields require the 'replace' option. "
+                             "")
```

Comment:
Yo could mention that protected metadata blocks can not be edited with this tool. For that you would need to have the key as input and pas it to the dataverse api.